src/phylum/ci/ci_none.py

```diff
@@ -45,12 +45,20 @@ def common_ancestor_commit(self) -> Optional[str]:
         remote = git_remote()
         cmd = ["git", "merge-base", "HEAD", f"refs/remotes/{remote}/HEAD"]
         try:
-            common_commit = subprocess.run(cmd, check=True, capture_output=True, text=True).stdout.strip()  # noqa: S603
-        except subprocess.CalledProcessError as err:
-            pprint_subprocess_error(err)
-            LOG.warning("The common ancestor commit could not be found")
-            common_commit = None
-        return common_commit
+            commit = subprocess.run(cmd, check=True, capture_output=True, text=True).stdout.strip()  # noqa: S603
+        except subprocess.CalledProcessError as outer_err:
+            # The most likely problem is that the remote HEAD ref is not set. The attempt to set it here, inside
+            # the except block, is due to wanting to minimize calling commands that require git credentials.
+            pprint_subprocess_error(outer_err)
+            LOG.warning("Failed to get commit. Remote HEAD ref likely not set. Attempting to set it and try again ...")
+            git_set_remote_head(remote)
```

Comment:
Is there no way to tell if the outer error is actually due to the remote HEAD ref not being set?

Reply:
It _might_ be possible...but likely not worth trying all the possibilities to narrow down the root cause of the error. The return code for `git merge-base` will be 128 when there is an error...regardless of the error. It doesn't distinguish between various error cases. For instance, the HEAD ref file (e.g., `.git/refs/remotes/origin/HEAD`) may not exist or it may exist but have an invalid entry (e.g., `ref: refs/remotes/origin/master` instead of `ref: refs/remotes/origin/main` where `origin/master` does not exist). Each of these error cases has a different error message, but it seems like overkill to look up all the possible cases and account for them in order to provide a better explanation of what went wrong.

Maybe more importantly, the command used in `git_set_remote_head` - `git remote set-head <remote> --auto` - will not complain if the remote HEAD ref is already set.

Reply:
I figured that might be the case. Thanks for the explanation, Charlie 👍 

I don't know if I have the context to be an approver here. Maybe @cd-work or @kylewillmon can give another review :)